src/modules/modifierOSites/process.cpp

```diff
@@ -63,12 +90,31 @@ Module::ExecutionResult ModifierOSitesModule::process(ModuleContext &moduleConte
         for (auto &&[oSite, index] : nearO)
         {
             oxygenSitesHistogram.bin(neighbourMap[oSite].size());
+
+            switch (neighbourMap[oSite].size())
+            {
+                case 0:
+                    histMFO.bin(targetConfiguration_->box()->minimumDistance(siteM->origin(), oSite->origin()));
+                    break;
+                case 1:
+                    histMNBO.bin(targetConfiguration_->box()->minimumDistance(siteM->origin(), oSite->origin()));
+                    break;
+                case 2:
+                    histMBO.bin(targetConfiguration_->box()->minimumDistance(siteM->origin(), oSite->origin()));
+                    break;
+                default:
+                    histMOtherO.bin(targetConfiguration_->box()->minimumDistance(siteM->origin(), oSite->origin()));
+            }
```

Comment:
...you can just do this!
```suggestion
            histogramsMO[std::min(neighbourMap[oSite].size(), 3)].bin(targetConfiguration_->box()->minimumDistance(siteM->origin(), oSite->origin()));
```